### ISSUE TITLE
Python program not terminating properly

### ISSUE THREAD

**Issue by kubatjak** [osmcode/pyosmium#234]
### Discussed in https://github.com/osmcode/pyosmium/discussions/233

<div type='discussions-op-text'>

<sup>Originally posted by **kubatjak** September  8, 2023</sup>
Hello,

I have been running a very simple python script using the SimpleHandler. The program runs correctly till the end but then it fails to terminate properly. The program cannot be killed by keyboard interupt. It happens in Windows command line and PowerShell and GitBash consoles. It happens only after using apply_file, just having an instance of Handler does not cause any issues.

Here's the script:
```python
import osmium
import atexit


class Handler(osmium.SimpleHandler):
    def __init__(self) -> None:
        super().__init__()
        self.num_nodes = 0

    def node(self, n):
        self.num_nodes += 1


def cleanup():
    print('Cleanup called')
    print('Trying to terminate the program ...')


def main():
    osm_file = r'map_14.4086_50.0726_14.4195_50.0761.osm'
    handler = Handler()
    handler.apply_file(osm_file)
    print('# nodes:', handler.num_nodes)


if __name__ == '__main__':
    atexit.register(cleanup)
    try:
        main()
    except Exception as e:
        print(e)
```

Would someone know what could be the cause of the problem?

Thank you!</div>

**Comment by joto** [comment]
Closed as duplicate of #233
